graphql/resolver.go

```diff
@@ -35,6 +36,46 @@ func New(apiURL string) Config {
 			sc: dbConnector,
 		},
 	}
+	c.Directives.RequireHostAccess = func(ctx context.Context, obj interface{}, next graphql.Resolver, access HostAccessLevel) (interface{}, error) {
+		args, isStringMap := obj.(map[string]interface{})
+		if !isStringMap {
+			return nil, ResourceNotFound.Send(ctx, "host not specified")
+		}
+		hostId, hasHostId := args["hostId"].(string)
+		hostIdsInterface, hasHostIds := args["hostIds"].([]interface{})
+		hostIds := []string{}
+		if hasHostIds {
+			for _, v := range hostIdsInterface {
+				hostIds = append(hostIds, v.(string))
+			}
+		}
+		if !hasHostId && !hasHostIds {
+			return nil, ResourceNotFound.Send(ctx, "host not specified")
+		}
+		hostIdsToCheck := []string{}
+		hostIdsToCheck = append(hostIdsToCheck, hostId)
+		hostIdsToCheck = append(hostIdsToCheck, hostIds...)
+		var requiredLevel int
+		if access == HostAccessLevelEdit {
+			requiredLevel = evergreen.HostsEdit.Value
+		} else if access == HostAccessLevelView {
+			requiredLevel = evergreen.HostsView.Value
+		}
+		user := mustHaveUser(ctx)
+		hostsToCheck, err := host.Find(ctx, host.ByIds(hostIdsToCheck))
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, "Error getting hosts")
+		}
+		if len(hostsToCheck) == 0 {
+			return nil, ResourceNotFound.Send(ctx, "No matching hosts found")
+		}
+		for _, h := range hostsToCheck {
+			if !userHasHostPermission(user, h.Distro.Id, requiredLevel) {
```

Comment:
This should be consistent with the logic [here](https://github.com/evergreen-ci/evergreen/pull/8337/files#diff-18582cc382c2280a0446b8fa448fd4ca997753d6ebfe6042a981701b312164c6R344) (as in whether we keep track of a list of forbidden hosts or not)